onnx/test/shape_inference_test.py

```diff
@@ -189,64 +233,79 @@ def _identity_prop(self, op: str, **kwargs: Any) -> None:
             graph, [make_tensor_value_info("y", TensorProto.FLOAT, (30, 4, 5))]
         )
 
-    def test_transpose(self) -> None:
+    @parameterized.expand(all_versions_for("Transpose"))
+    def test_transpose(self, _, version) -> None:
         graph = self._make_graph(
             [("X", TensorProto.FLOAT, (2, 3, 4))],
             [make_node("Transpose", ["X"], ["Y"], perm=[1, 0, 2])],
             [],
         )
         self._assert_inferred(
-            graph, [make_tensor_value_info("Y", TensorProto.FLOAT, (3, 2, 4))]
+            graph,
+            [make_tensor_value_info("Y", TensorProto.FLOAT, (3, 2, 4))],
+            opset_imports=[helper.make_opsetid(ONNX_DOMAIN, version)],
```

Comment:
nit: would it be cleaner to make `_assert_inferred` handle `opset_imports=[helper.make_opsetid(ONNX_DOMAIN, version)]` and then we can just pass version to it?

Reply:
It would be difficult to do so, because `opset_imports` does not only handler versions, it also handles domains. For example, in `test_category_mapper`, the `opset_imports` is fed with different version as well as different domains.
```python
        opset_imports=[
                make_opsetid(ONNX_ML_DOMAIN, 1),
                make_opsetid(ONNX_DOMAIN, 11),
            ],
```